### ISSUE TITLE
Singularity to recognize `oras://`

### ISSUE THREAD

**Issue by gaow** [vatlab/sos#1524]
Currently we support automatically select container engine for the following: https://vatlab.github.io/sos-docs/doc/user_guide/h.html#Options-container-and-engine . It would be great if we can add more for `singularity`. According to:

```
$ singularity exec -h
Run a command within a container

Usage:
  singularity exec [exec options...] <container> <command>

Description:
  singularity exec supports the following formats:

  *.sif               Singularity Image Format (SIF). Native to Singularity 3.0+

  *.sqsh              SquashFS format.  Native to Singularity 2.4+

  *.img               ext3 format. Native to Singularity versions < 2.4.

  directory/          sandbox format. Directory containing a valid root file
                      system and optionally Singularity meta-data.

  instance://*        A local running instance of a container. (See the instance
                      command group.)

  library://*         A container hosted on a Library (default
                      https://cloud.sylabs.io/library)

  docker://*          A container hosted on Docker Hub

  shub://*            A container hosted on Singularity Hub

  oras://*            A container hosted on a supporting OCI registry
```

the one we need specifically now is `oras://` that we currently dont support.

**Comment by BoPeng** [comment]
> I should mention that adding engine="singularity" does not work:

So we should at least make the file format open. I think.

**Comment by BoPeng** [comment]
Please test if the `orab` branch works.

**Comment by gaow** [comment]
Thank you @BoPeng , this script:

```
[default]
bash: container="oras://ghcr.io/cumc/bioinfo_apptainer:latest", engine="singularity"
  ls /opt/conda/envs/
```

works perfectly. I think it is ready to merge to `master`. Thank you!

**Comment by BoPeng** [comment]
Merged and released sos 0.24.3.
